app/livechat/server/publications/livechatRooms.js

```diff
@@ -9,7 +9,7 @@ Meteor.publish('livechat:rooms', function(filter = {}, offset = 0, limit = 20) {
 		return this.error(new Meteor.Error('error-not-authorized', 'Not authorized', { publish: 'livechat:rooms' }));
 	}
 
-	if (!hasPermission(this.userId, 'view-livechat-rooms')) {
+	if (!hasPermission(this.userId, 'view-livechat-rooms') && !(filter._id && hasPermission(this.userId, 'view-l-room'))) {
```

Comment:
also is this the case where an agent is trying to view a room's data? in this case should we only allow it to view the data if he is the `servedBy` agent?

Reply:
Yeah, this is exactly that case.
I just think that since we're extending Livechat permissions, as well as creating new Livechat Roles, maybe there will be some cases where another user other than an agent will have the same permission to access the room, such as a Livechat Monitor.
Due to that, I didn't restrict those validations to accept only the agent who is serving the room, you know?

Thanks.